PWGHF/vertexingHF/AliAnalysisTaskSEHFSystPID.cxx

```diff
@@ -1329,3 +1383,86 @@ void AliAnalysisTaskSEHFSystPID::TagOOBPileUpEvent() {
+
+ //________________________________________________________________
+bool AliAnalysisTaskSEHFSystPID::IsSelectedOmega(AliAODcascade *const casc)
+{
+  if (!casc) return false;
+
+  double vtxCasc[3];
+
+  vtxCasc[0] = casc->DecayVertexXiX();
+  vtxCasc[1] = casc->DecayVertexXiY();
+  vtxCasc[2] = casc->DecayVertexXiZ();
+
+  float cascRad = TMath::Sqrt(vtxCasc[0] * vtxCasc[0] + vtxCasc[1] * vtxCasc[1]);
+  if (cascRad < fCutMinCascRadius) return false;
+
+  float cascV0Rad = casc->RadiusSecVtx();
+  if (cascV0Rad < fCutMinV0Radius) return false;
+  if (cascV0Rad > fCutMaxV0Radius) return false;
+
+  double PV[3];
+  const AliAODVertex *vertex = fAOD->GetPrimaryVertex();
+  if(!vertex) return false;
+  vertex->GetXYZ(PV);
+
+  double cascCosPA = casc->CosPointingAngleXi((const double&) PV[0], (const double&) PV[1], (const double&) PV[2]);
+  if (cascCosPA < fCutCosPA) return false;
+
+  double v0CosPA = casc->CosPointingAngle(PV); 
+  if (v0CosPA < fCutCosPA) return false;
+
+  AliAODTrack* pTrack=dynamic_cast<AliAODTrack*>(casc->GetDaughter(0));
+  AliAODTrack* nTrack=dynamic_cast<AliAODTrack*>(casc->GetDaughter(1));
+  AliAODTrack* bTrack=dynamic_cast<AliAODTrack*>(casc->GetDecayVertexXi()->GetDaughter(0));
+
+  float cascDcaBachToPV = casc->DcaBachToPrimVertex();
+  if (cascDcaBachToPV < fCutDcaBachToPV) return false;
+
+  float cascDcaV0ToPV = casc->DcaV0ToPrimVertex();
+  if (cascDcaV0ToPV < fCutDcaV0ToPV) return false;
+
+  float cascDcaPosToPV = casc->DcaPosToPrimVertex();
+  float cascDcaNegToPV = casc->DcaNegToPrimVertex();
+
+  if (cascDcaPosToPV < fCutDcaV0DaughToPV) return false;
+  if (cascDcaNegToPV < fCutDcaV0DaughToPV) return false;
+ 
+  
+  float cascDcaV0Daught = casc->DcaV0Daughters();
+  if(cascDcaV0Daught > fDcaV0Daught) return false;
+
+  float cascDcaCascDaught = casc->DcaXiDaughters();
+  if(cascDcaCascDaught > fDcaCascDaught) return false;
+
+  float cascInvMassLam;
+  if(casc->ChargeXi() < 0){
+    cascInvMassLam = casc->MassLambda();
+  } else {
+    cascInvMassLam = casc->MassAntiLambda();
+  }
+  if(TMath::Abs(cascInvMassLam - kLambdaMass) > fCutInvMassLam) return false;
+
+  float cascNSigPosProton = fPIDresp->NumberOfSigmasTPC(pTrack, AliPID::kProton);
+  float cascNSigPosPion = fPIDresp->NumberOfSigmasTPC(pTrack, AliPID::kPion);
+  float cascNSigNegProton = fPIDresp->NumberOfSigmasTPC(nTrack, AliPID::kProton);
+  float cascNSigNegPion = fPIDresp->NumberOfSigmasTPC(nTrack, AliPID::kPion);
+  float cascNSigBacKaon = fPIDresp->NumberOfSigmasTPC(bTrack, AliPID::kKaon);
+
+  if(casc->ChargeXi() < 0){
+   if (TMath::Abs(cascNSigNegPion) > fCutNSigmaPID   || 
+       TMath::Abs(cascNSigPosProton) > fCutNSigmaPID ||
+       TMath::Abs(cascNSigBacKaon) > fCutNSigmaPID){
+         return false;
+       }
+  } else {
+    if(TMath::Abs(cascNSigPosPion) > fCutNSigmaPID   ||
+       TMath::Abs(cascNSigNegProton) > fCutNSigmaPID || 
+       TMath::Abs(cascNSigBacKaon) > fCutNSigmaPID){
+         return false;
+       }
+    }
```

Comment:
I would suggest to remove this: the idea of the task is to select strange particles without using the PID information, otherwise the PID distributions cannot be studied anymore